app/grandchallenge/publications/fields.py

```diff
@@ -0,0 +1,124 @@
+import re
+
+from django.core.validators import RegexValidator
+from django.db import models
+from django.forms import CharField
+
+from grandchallenge.publications.utils import get_doi_csl
+from grandchallenge.publications.utils.manubot import get_arxiv_csl
+
+# regex modified for python syntax from
+# https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+DOI_REGEX = r"^10\.\d{4,9}/[-._;()/:a-z0-9]+$"
+ARXIV_REGEX = r"^\d{4}\.\d{4,5}$"
+
+identifier_validator = RegexValidator(regex=f"{DOI_REGEX}|{ARXIV_REGEX}")
+
+
+class PublicationType(models.TextChoices):
+    DOI = "D"
+    ARXIV = "A"
+
+
+class PublicationIdentifier:
+    def __init__(self, identifier):
+        self.identifier = identifier

```

Comment:
It would be nicer to make this protected as you don't really want it to be changed outside of the class.

```suggestion
    def __init__(self, identifier):
        self.__identifier = identifier
```